frontend/src/components/pages/RegistrantExperience/RegistrationSteps/AdditionalInfo/index.tsx

```diff
@@ -1,26 +1,90 @@
-import React from "react";
-import { Box, Checkbox, Text } from "@chakra-ui/react";
+import { Box, Divider, Text, VStack } from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import { RegistrantExperienceCamper } from "../../../../../types/CamperTypes";
+import { FormQuestion } from "../../../../../types/CampsTypes";
+import { useAdditionalInfoDispatcher } from "./additionalInfoReducer";
+import CamperQuestionsCard from "./QuestionCards/CamperQuestionsCard";
+import EarlyDropOffLatePickupCard from "./QuestionCards/EarlyDropOffLatePickupCard";
 
 type AdditionalInfoProps = {
-  isChecked: boolean;
-  toggleChecked: () => void;
+  nextBtnRef: React.RefObject<HTMLButtonElement>;
+  campers: RegistrantExperienceCamper[];
+  setCampers: React.Dispatch<
+    React.SetStateAction<RegistrantExperienceCamper[]>
+  >;
+  campName: string;
+  formQuestions: FormQuestion[];
+  hasEarlyDropOffLatePickup: boolean;
+  requireEarlyDropOffLatePickup: boolean | null;
+  setRequireEarlyDropOffLatePickup: React.Dispatch<
+    React.SetStateAction<boolean | null>
+  >;
 };
 
 const AdditionalInfo = ({
-  isChecked,
-  toggleChecked,
+  nextBtnRef,
+  campers,
+  setCampers,
+  campName,
+  formQuestions,
+  hasEarlyDropOffLatePickup,
+  requireEarlyDropOffLatePickup,
+  setRequireEarlyDropOffLatePickup,
 }: AdditionalInfoProps): React.ReactElement => {
+  const dispatchAdditionalInfoAction = useAdditionalInfoDispatcher(setCampers);
+
+  const [submitClicked, setSubmitClicked] = useState(false);
+
+  useEffect(() => {
+    let nextBtnRefValue: HTMLButtonElement; // Reference to the next step button
+
+    if (nextBtnRef && nextBtnRef.current) {
+      nextBtnRefValue = nextBtnRef.current;
+      nextBtnRefValue.addEventListener("click", () => setSubmitClicked(true));
+    }
+
+    return () => {
+      if (nextBtnRefValue) {
+        nextBtnRefValue.removeEventListener("click", () =>
+          setSubmitClicked(true),
+        );
+      }
+    };
+  }, [campers, nextBtnRef]);
+
   return (
-    <Box>
-      <Text textStyle="displayXLarge">Camp Registration</Text>
-      <Text>Additional Info</Text>
-
-      <Checkbox
-        size="lg"
-        borderColor="black"
-        isChecked={isChecked}
-        onChange={toggleChecked}
-      />
+    <Box pb={14}>
+      <Text textStyle="displayXLarge">{`${campName} Registration`}</Text>
+      <VStack alignItems="flex-start" spacing={8} marginTop={8}>
+        <Text textStyle="displayLarge" textColor="primary.green.100">
+          Camper-specific Additional Questions
+        </Text>
+        {campers.map((camper, camperIndex) => (
+          <CamperQuestionsCard
+            key={camperIndex}
```

Comment:
Can we make the key more descriptive? eg: `additional_info_camper_${camperIndex}`

Reply:
could you make the keys in other components more descriptive too? Not a major blocker but otherwise it becomes really hard to track keys